lntest/itest/lnd_onchain_test.go

```diff
@@ -379,3 +382,237 @@ func testAnchorReservedValue(net *lntest.NetworkHarness, t *harnessTest) {
+	defer shutdownAndAssert(net, t, alice)
+
+	bob := net.NewNode(t.t, "Bob", args)
+	defer shutdownAndAssert(net, t, bob)
+
+	ctxb := context.Background()
+	net.ConnectNodes(t.t, alice, bob)
+
+	// We'll fund our Alice with coins, as she'll be opening the channel.
+	// We'll fund her with *just* enough coins to open the channel.
+	const (
+		firstChanSize  = 1_000_000
+		secondChanSize = 500_000
+	)
+	net.SendCoins(t.t, firstChanSize+secondChanSize, alice)
+
+	// We'll give Alice another spare UTXO as well so she can use it to
+	// help sweep all coins.
+	net.SendCoins(t.t, secondChanSize, alice)
+
+	// Open the channel between the two nodes and wait for it to confirm
+	// fully.
+	aliceChanPoint1 := openChannelAndAssert(
+		t, net, alice, bob, lntest.OpenChannelParams{
+			Amt: firstChanSize,
+		},
+	)
+
+	// With the channel open, we'll actually immediately force close it. We
+	// don't care about network announcements here since there's no routing
+	// in this test.
+	_, _, err := net.CloseChannel(alice, aliceChanPoint1, true)
+	if err != nil {
+		t.Fatalf("unable to execute force channel closure: %v", err)
+	}
+
+	// Now that the channel has been force closed, it should show up in the
+	// PendingChannels RPC under the waiting close section.
+	pendingChansRequest := &lnrpc.PendingChannelsRequest{}
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	pendingChanResp, err := alice.PendingChannels(ctxt, pendingChansRequest)
+	if err != nil {
+		t.Fatalf("unable to query for pending channels: %v", err)
+	}
+	err = checkNumWaitingCloseChannels(pendingChanResp, 1)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	// Get the normal channel outpoint so we can track it in the set of
+	// channels that are waiting to be closed.
+	fundingTxID, err := lnrpc.GetChanPointFundingTxid(aliceChanPoint1)
+	if err != nil {
+		t.Fatalf("unable to get txid: %v", err)
+	}
+	chanPoint := wire.OutPoint{
+		Hash:  *fundingTxID,
+		Index: aliceChanPoint1.OutputIndex,
+	}
+	waitingClose, err := findWaitingCloseChannel(pendingChanResp, &chanPoint)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	// At this point, the channel is waiting close, and we have both the
+	// commitment transaction and anchor sweep in the mempool.
+	const expectedTxns = 2
+	sweepTxns, err := getNTxsFromMempool(
+		net.Miner.Client, expectedTxns, minerMempoolTimeout,
+	)
+	require.NoError(t.t, err, "no sweep txns in miner mempool")
+	aliceCloseTx := waitingClose.Commitments.LocalTxid
+	_, aliceAnchor := findCommitAndAnchor(t, net, sweepTxns, aliceCloseTx)
+
+	// We'll now mine _only_ the commitment force close transaction, as we
+	// want the anchor sweep to stay unconfirmed.
+	var emptyTime time.Time
+	forceCloseTxID, _ := chainhash.NewHashFromStr(aliceCloseTx)
+	commitTxn, err := net.Miner.Client.GetRawTransaction(
+		forceCloseTxID,
+	)
+	if err != nil {
+		t.Fatalf("unable to get transaction: %v", err)
+	}
+	_, err = net.Miner.GenerateAndSubmitBlock(
```

Comment:
damn, TIL!!! So I guess you can generate an empty block too? Asking because I had issues writing tests for fee bumper, as I wanted to generate blocks without mining the txes to cause a fee bump action, ended up in this weird [workaround](https://github.com/lightninglabs/nautilus/pull/745/commits/b282c9df9bdc6ca917271b92595a556648d347c6#diff-c14c40b21d7dc21dbb27d62169e7aa2a9dcb747bf53fd556a67bfecdf0f49d68R451), should have done my research more!!

Reply:
Yeh exactly (just pass nil values there), we added this call when working on some anchor related itests in the past. It can also be useful to simulate stuff like transaction pinning, unexpected confirmation delays, etc, etc.